File: oldSnacks/python/MBTIPersonalityTestFunction.py

extrovertVsintrovert = [];
AE = 0;
BI = 0;
sensingVsintuitive = [];
AS = 0;
BN = 0;
thinkingVsfeeling = [];
AT = 0;
BF = 0;
judgingVsperceptive = [];
AJ = 0;
BP = 0;
result = "";


def extrovertORintrovert(choosed, number):
	if(number == 1):
		extrovertVsintrovert.append(choosed);
	if(number == 5):
		extrovertVsintrovert.append(choosed);
	if(number == 9):
		extrovertVsintrovert.append(choosed);
	if(number == 13):
		extrovertVsintrovert.append(choosed);
	if(number == 17):
		extrovertVsintrovert.append(choosed);

	return extrovertVsintrovert;



def extrovertORintrovertCounter(answer):
	global AE
	global BI
	if(answer == "A"):
		AE = AE + 1;
	if(answer == "B"):
		BI = BI + 1;
	return AE, BI;

def sensingORintuitive(choosed, number):
	if(number == 2):
		sensingVsintuitive.append(choosed);
	if(number == 6):
		sensingVsintuitive.append(choosed);
	if(number == 10):
		sensingVsintuitive.append(choosed);
	if(number == 14):
		sensingVsintuitive.append(choosed);
	if(number == 18):
		sensingVsintuitive.append(choosed);

	return sensingVsintuitive;



def sensingORintuitiveCounter(answer):
	global AS
	global BN
	if(answer == "A"):
		AS = AS + 1;
	if(answer == "B"):
		BN = BN + 1;
	return AS, BN;



def thinkingORfeeling(choosed, number):
	if(number == 3):
		thinkingVsfeeling.append(choosed);
	if(number == 7):
		thinkingVsfeeling.append(choosed);
	if(number == 11):
		thinkingVsfeeling.append(choosed);
	if(number == 15):
		thinkingVsfeeling.append(choosed);
	if(number == 19):
		thinkingVsfeeling.append(choosed);

	return thinkingVsfeeling;



def thinkingORfeelingCounter(answer):
	global AT
	global BF
	if(answer == "A"):
		AT = AT + 1;
	if(answer == "B"):
		BF = BF + 1;
	return AT, BF;


def judgingORperceptive(choosed, number):
	if(number == 4):
		judgingVsperceptive.append(choosed);
	if(number == 8):
		judgingVsperceptive.append(choosed);
	if(number == 12):
		judgingVsperceptive.append(choosed);
	if(number == 16):
		judgingVsperceptive.append(choosed);
	if(number == 20):
		judgingVsperceptive.append(choosed);

	return judgingVsperceptive;



def judgingORperceptiveCounter(answer):
	global AJ
	global BP
	if(answer == "A"):
		AJ = AJ + 1;
	if(answer == "B"):
		BP = BP + 1;
	return AJ, BP;

def extrovertORletrovertAnswer():
	for counter in extrovertVsintrovert:
		print(counter);

	print(f"Numbers of A selected: {AE}");
	print(f"Numbers of B selected: {BI}");
	print();



def sensingORintuitiveAnswer():
	for counter in sensingVsintuitive:
		print(counter);

	print(f"Numbers of A selected: {AS}");
	print(f"Numbers of B selected: {BN}");
	print();



def thinkingORfeelingAnswer():
	for counter in thinkingVsfeeling:
		print(counter);

	print(f"Numbers of A selected: {AT}");
	print(f"Numbers of B selected: {BF}");
	print();



def judgingORperceptiveAnswer():
	for counter in judgingVsperceptive:
		print(counter);

	print(f"Numbers of A selected: {AJ}");
	print(f"Numbers of B selected: {BP}");
	print();


def overallResult():
	result = "";
	if(AE > BI):
		result += "E";
	else:
		result += "I";


	if(AS > BN):
		result += "S";
	else:
		result += "N";


	if(AT > BF):
		result += "T";
	else:
		result += "F";


	if(AJ > BP):
		result += "J";
	else:
		result += "P";

	return result;


#public static void overallResultType(){

File: oldSnacks/java/MBTIPersonalityTestFunction.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Arrays;

public class MBTIPersonalityTestFunction{

static String[] extrovertVsintrovert = new String[5];
static int AE = 0;
static int BI = 0;
static String[] sensingVsintuitive = new String[5];
static int AS = 0;
static int BN = 0;
static String[] thinkingVsfeeling = new String[5];
static int AT = 0;
static int BF = 0;
static String[] judgingVsperceptive = new String[5];
static int AJ = 0;
static int BP = 0;
static String result = "";


public static String[] extrovertORintrovert(String choosed, int number){
if(number == 1) extrovertVsintrovert[0] = choosed;
if(number == 5) extrovertVsintrovert[1] = choosed;
if(number == 9) extrovertVsintrovert[2] = choosed;
if(number == 13) extrovertVsintrovert[3] = choosed;
if(number == 17) extrovertVsintrovert[4] = choosed;

return extrovertVsintrovert;
}


public static void extrovertORintrovertCounter(String answer){
if(answer.equals("A")) AE++;
if(answer.equals("B")) BI++;
}

public static String[] sensingORintuitive(String choosed, int number){
if(number == 2) sensingVsintuitive[0] = choosed;
if(number == 6) sensingVsintuitive[1] = choosed;
if(number == 10) sensingVsintuitive[2] = choosed;
if(number == 14) sensingVsintuitive[3] = choosed;
if(number == 18) sensingVsintuitive[4] = choosed;

return sensingVsintuitive;
}


public static void sensingORintuitiveCounter(String answer){
if(answer.equals("A")) AS++;
if(answer.equals("B")) BN++;
}


public static String[] thinkingORfeeling(String choosed, int number){
if(number == 3) thinkingVsfeeling[0] = choosed;
if(number == 7) thinkingVsfeeling[1] = choosed;
if(number == 11) thinkingVsfeeling[2] = choosed;
if(number == 15) thinkingVsfeeling[3] = choosed;
if(number == 19) thinkingVsfeeling[4] = choosed;

return thinkingVsfeeling;
}


public static void thinkingORfeelingCounter(String answer){
if(answer.equals("A")) AT++;
if(answer.equals("B")) BF++;
}


public static String[] judgingORperceptive(String choosed, int number){
if(number == 4) judgingVsperceptive[0] = choosed;
if(number == 8) judgingVsperceptive[1] = choosed;
if(number == 12) judgingVsperceptive[2] = choosed;
if(number == 16) judgingVsperceptive[3] = choosed;
if(number == 20) judgingVsperceptive[4] = choosed;

return judgingVsperceptive;
}


public static void judgingORperceptiveCounter(String answer){
if(answer.equals("A")) AJ++;
if(answer.equals("B")) BP++;
}

public static void extrovertORintrovertAnswer(){
for(int counter = 0; counter < extrovertVsintrovert.length; counter++){
System.out.println(extrovertVsintrovert[counter]);
}
System.out.println("Numbers of A selected: "+AE);
System.out.println("Numbers of B selected: "+BI);
System.out.println();
}


public static void sensingORintuitiveAnswer(){
for(int counter = 0; counter < sensingVsintuitive.length; counter++){
System.out.println(sensingVsintuitive[counter]);
}
System.out.println("Numbers of A selected: "+AS);
System.out.println("Numbers of B selected: "+BN);
System.out.println();
}


public static void thinkingORfeelingAnswer(){
for(int counter = 0; counter < thinkingVsfeeling.length; counter++){
System.out.println(thinkingVsfeeling[counter]);
}
System.out.println("Numbers of A selected: "+AT);
System.out.println("Numbers of B selected: "+BF);
System.out.println();
}


public static void judgingORperceptiveAnswer(){
for(int counter = 0; counter < judgingVsperceptive.length; counter++){
System.out.println(judgingVsperceptive[counter]);
}
System.out.println("Numbers of A selected: "+AJ);
System.out.println("Numbers of B selected: "+BP);
System.out.println();
}

public static String overallResult(){
if(AE > BI){
result += "E";
}else{
result += "I";
}

if(AS > BN){
result += "S";
}else{
result += "N";
}

if(AT > BF){
result += "T";
}else{
result += "F";
}

if(AJ > BP){
result += "J";
}else{
result += "P";
}
return result;
}

//public static void overallResultType(){


}